File: lib/test/t_LinearModelFactory_std.cxx
//                                               -*- C++ -*-
/**
 *  @brief The test file of class LinearModelFactory
 *
 *  Copyright 2005-2016 Airbus-EDF-IMACS-Phimeca
 *
 *  This library is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU Lesser General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
#include "openturns/OT.hxx"
#include "openturns/OTtestcode.hxx"

using namespace OT;
using namespace OT::Test;

int main(int argc, char *argv[])
{
  TESTPREAMBLE;
  OStream fullprint(std::cout);
  setRandomGenerator();

  /* Default constructor */
  LinearModelFactory lmfact;

  /* lm build */
  UnsignedInteger size = 20;
  NumericalSample oneSample(size, 1);
  NumericalSample twoSample(size, 1);
  for (UnsignedInteger i = 0; i < size; ++i)
  {
    oneSample[i][0] = 7.0 * sin(-3.5 + (6.5 * i) / (size - 1.0)) + 2.0;
    twoSample[i][0] = -2.0 * oneSample[i][0] + 3.0 + 0.05 * sin(oneSample[i][0]);
  }
  LinearModel test = lmfact.build(oneSample, twoSample);
  fullprint << "test = " << test << std::endl;

  return ExitCode::Success;
}

File: python/test/t_LinearModelFactory_std.py
#! /usr/bin/env python

from __future__ import print_function
from openturns import *
from math import *

TESTPREAMBLE()
RandomGenerator.SetSeed(0)

try:

    # TEST NUMBER ZERO : DEFAULT CONSTRUCTOR AND CREATION OF LM
    print(
        "test number zero : default constructor and creation of linear model")

    # Default constructor
    lmfact = LinearModelFactory()

    # lm build
    size = 20
    oneSample = Sample(size, 1)
    twoSample = Sample(size, 1)
    for i in range(size):
        oneSample[i] = Point(
            1, 7.0 * sin(-3.5 + (6.5 * i) / (size - 1.0)) + 2.0)
        twoSample[i] = Point(
            1, -2.0 * oneSample[i, 0] + 3.0 + 0.05 * sin(oneSample[i, 0]))
    test = lmfact.build(oneSample, twoSample)
    print("test = ", repr(test))

except:
    import sys
    print("t_LinearModelFactory_std_std.py",
          sys.exc_info()[0], sys.exc_info()[1])
